### PR TITLE
pr for microverse tse approval

### PR DESCRIPTION
**Pull request by firmoholanda**

### CODE REVIEW

**Comment by shshamim63** [comment]
## Status :ambulance: [!! invalid !!] :ambulance: 
___
Hi @firmoholanda :wave: ,
I am really sorry to declare this review request as invalid :disappointed: . The main two reason for declaring a review request are below.
1. PR don't contain a short description about the task that has been covered in this PR.
2. Did not follow Gitflow. 
image
___
### Kindly fix the issue and send a review request.
___
## :gear: Happy Coding :gear: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/) | [Slack](@Shakhawat)

File: ruby_enumerable_methods.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    for item in self do
+      yield(item)
+    end
+  end
```
Comment (uche-inyama) [inline ruby_enumerable_methods.rb:8]: - enumerables my_each should return an Enumerator if no block is given
     Failure/Error: expect(array.my_each).to be_an(Enumerator)

File: ruby_enumerable_methods.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  def my_each_with_index
+    i = 0
+    for item in self do
+      yield(item, i)
+      i += 1
+    end
+  end
```
Comment (uche-inyama) [inline ruby_enumerable_methods.rb:16]: - enumerables my_each_with_index returns an enumerator if no block is given
     Failure/Error: expect(array.my_each_with_index).to be_an(Enumerator)

File: ruby_enumerable_methods.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+  def my_select
+    select_arr = []
+    for item in self do
+      select_arr << item if yield(item)
+    end
+    select_arr
+  end
```
Comment (uche-inyama) [inline ruby_enumerable_methods.rb:24]: - enumerables my_select returns an enumerator if no block is given
     Failure/Error: expect(array.my_select).to be_an(Enumerator)

File: ruby_enumerable_methods.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+  def my_all?
+    pass = false
+    for item in self do
+      pass = yield(item)
+      break unless pass
+    end
+    pass
+  end
```
Comment (uche-inyama) [inline ruby_enumerable_methods.rb:33]: - enumerables my_all? when no block or argument is given returns true when none of the collection members are false or nil
     Failure/Error: expect(true_array.my_all?).to be true_array.all?

- enumerables my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
     Failure/Error: expect(array.my_all?(Integer)).to be array.all?(Integer)

- enumerables my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
     Failure/Error: expect(words.my_all?(/d/)).to be words.all?(/d/)

- enumerables my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
     Failure/Error: expect(array.my_all?(3)).to be array.all?(3)

File: ruby_enumerable_methods.rb [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+  def my_any?
+    pass = false
+    for item in self do
+      pass = yield(item)
+      break if pass
+    end
+    pass
+  end
```
Comment (uche-inyama) [inline ruby_enumerable_methods.rb:42]: - enumerables my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
     Failure/Error: expect(true_array.my_any?).to be true_array.any?

- enumerables my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
     Failure/Error: expect(array.my_any?(Integer)).to be array.any?(Integer)

-  enumerables my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
      Failure/Error: expect(words.my_any?(/z/)).to be words.any?(/z/)

- enumerables my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
      Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')

File: ruby_enumerable_methods.rb [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+  def my_none?
+    pass = false
+    for item in self do
+      pass = !yield(item)
+      break unless pass
+    end
+    pass
+  end
```
Comment (uche-inyama) [inline ruby_enumerable_methods.rb:51]: - enumerables my_none? when no block or argument is given returns true only if none of the collection members is true
      Failure/Error: expect(false_array.my_none?).to be true

- enumerables my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
      Failure/Error: expect(array.my_none?(String)).to be true

-  enumerables my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
      Failure/Error: expect(words.my_none?(/z/)).to be words.none?(/z/)
      
- enumerables my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)

File: ruby_enumerable_methods.rb [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+  def my_count
+    length
+  end
```
Comment (uche-inyama) [inline ruby_enumerable_methods.rb:55]: -  enumerables my_count counts the number of items in enum that are equal to item if an argument is given
      Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)

- enumerables my_count counts the number of elements yielding a true value if a block is given
      Failure/Error: expect(array.my_count(&block)).to eq array.count(&block)

File: ruby_enumerable_methods.rb [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+  def my_map(my_proc = nil)
+    new_arr = []
+    return self unless my_proc || block_given?
+
+    for item in self do
+      block_given? ? new_arr << yield(item) : new_arr << code.call(item)
+    end
+    new_arr
+  end
```
Comment (uche-inyama) [inline ruby_enumerable_methods.rb:65]: - enumerables my_map returns an Enumerator if no block is given
      Failure/Error: expect(array.my_map).to be_an(Enumerator)

File: ruby_enumerable_methods.rb [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+  def my_inject(num = nil)
+    accumulator = num.nil? ? first : num
+    my_each { |i| accumulator = yield(accumulator, i) }
+    accumulator
+  end
+end
```
Comment (uche-inyama) [inline ruby_enumerable_methods.rb:72]: - enumerables my_inject combines all elements of enum by applying a binary operation, specified by a block
      Failure/Error: expect(array.my_inject(&operation)).to eq array.inject(&operation)

- enumerables my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method
      Failure/Error: expect(array.my_inject(:+)).to eq array.inject(:+)

**Review by uche-inyama — COMMENTED** [review]
Hi @firmoholanda 

I have left comments about methods in your enumerable that needs a review/update.

Use the points I left in the comment as a reference to what need to be done. More so, I recommend you go through this [document](https://ruby-doc.org/core-2.6.5/Enumerable.html). Study the corresponding methods and how the testcases were implemented and do same while writing your methods.

Also, search on google, how to write a statement that returns an enumerator in ruby.

After implementing the suggested changes, submit another form for a code review.

#### ``` Uchechukwu Henry Inyama ```

``` Uchechukwu Henry Inyama ```

**Pull request closed without merging**